### PR TITLE
Sharing SDK update to ObjectElementAdapter.cs

### PR DESCRIPTION
**Pull request by HodgsonSDAS**
Added missing Element changed events to ObjectElementAdapter.cs
Also fixes https://github.com/Microsoft/HoloToolkit-Unity/issues/259

### CODE REVIEW

File: Assets/HoloToolkit/Sharing/Scripts/SDK/ObjectElementAdapter.cs [hunk 1]
```diff
@@ -11,61 +11,160 @@ namespace HoloToolkit.Sharing
...
-        public ObjectElementAdapter() { }
+        /// <summary>
+        /// The Object Elemet we're listening to.
+        /// </summary>
+        public ObjectElement ObjectElementRoot { get; private set; }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="ObjectElementAdapter"/> that creates a new Object Element and starts to listen to it.
+        /// </summary>
+        public ObjectElementAdapter()
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Sharing/Scripts/SDK/ObjectElementAdapter.cs]: I'm not sure if creating a new Object Element in the constructor is a good idea, but it seems to be the pattern in other test apps.  I'd like to get more feedback on this, if possible.

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Sharing/Scripts/SDK/ObjectElementAdapter.cs]: Yeah, this didn't seem right, so I removed it.

**Comment by NeerajW** [comment]
Looks ok to me. @jheuvel-ms any thoughts?

**Comment by jwittner** [comment]
lgtm.

**Pull request merged**